docker/deploy.sh

```diff
@@ -1,56 +1,127 @@
-#!/bin/bash
-
-echo "Initiating 1Password SCIM Bridge Deployment"
-
-run_docker_compose() {
-    mv ./scimsession ./docker/compose/scimsession
-
-    set -e # if the scimsession is already in ^, don't fail
-
-    cd docker/compose
-    ./generate-env.sh
-    mv ./scimsession ../../
-
-    read -p 'Please enter your domain name: ' domain_name
-
-    sed -i.bak s/{YOUR-DOMAIN-HERE}/$domain_name/g docker-compose.yml
-
-    docker-compose up --build -d
-    mv docker-compose.yml.bak docker-compose.yml
-
-    docker-compose logs -f
-}
-
-run_docker_swarm(){
-    mv ./scimsession ./docker/swarm/scimsession
-    cd docker/swarm
-    ./generate-secret.sh
-
-    set -e # if the secret already exists, don't fail
-
-    mv ./scimsession ../../
-
-    read -p 'Please enter your domain name: ' domain_name
-
-    sed -i.bak s/{YOUR-DOMAIN-HERE}/$domain_name/g docker-compose.yml
-
-    docker stack deploy -c docker-compose.yml op-scim
-    mv docker-compose.yml.bak docker-compose.yml
-
-    docker service logs --raw -f op-scim_scim
-}
-
-echo 'Please ensure you have read the README before continuing.'
-echo 'Which docker deployment method are you using?'
-read -p '[compose] or [swarm]: ' docker_path
-
-if [ "$docker_path" == "compose" ];
-then 
-    run_docker_compose
-elif [ "$docker_path" == "swarm" ]
+#!/usr/bin/env bash
+
+# Docker Swarm deployment script
+# Please ensure you've read PREPARING.md and docker/README.md
+
+# set the full path of the docker examples directory
+docker_path=$(dirname $(realpath $0))
+
+echo "Initiating 1Password SCIM Bridge Deployment to Docker Swarm"
+echo " "
+echo "Please specify the following options."
+
+while ! [[ "$docker_type" =~ ^(swarm|compose)$ ]]; do
+    read -p "Docker Swarm or Docker Compose? [swarm/compose]: " docker_type
+done
+
+read -p "Domain name you are deploying to [e.g: 'op-scim.example.com']: " domain_name
+
+while ! [[ -f "$scimsession_file" ]]; do
+    read -p "Path to your scimsession file: " scimsession_file
+    if ! [[ -f "$scimsession_file" ]]
+    then
+        echo "File '$scimsession_file' does not exist at that path, please try again." >&2
+    fi
+done
+
+echo " "
+echo "Using the following parameters to deploy the SCIM Bridge"
+echo "Deployment type:" $docker_type
+echo "scimsession file path:" $scimsession_file
+echo "Domain name:" $domain_name
+
+while ! [[ "$proceed" =~ ^([yY][eE][sS]|[yY])$ || "$proceed" =~ ^([nN][oO][nN]) ]]; do
+    read -p "Does this look correct? [Y/n]: " proceed
+    if [[ "$proceed" =~ ^([nN][oO][nN])$ ]]
+    then
+        echo "Exiting..."
+        exit 1
+    fi
+done
+
+# place the domain name into the deployment file, in a backup
+docker_file_path=$docker_path/$docker_type
+docker_original_file=$docker_file_path/docker-compose.yml
+docker_deploy_file=$docker_file_path/docker-compose.yml.deploy
```

Comment:
Just to be clear, having the ending not strictly be a `.yml` and instead be a `.yml.deploy` does not interfere with the docker tools, correct?

Reply:
Works just fine if you're specifying the file by explicit name.